src/adapters/world-name-permission-checker.ts

```diff
@@ -8,11 +8,34 @@ type NamesResponse = {
   nfts: { name: string; owner: { id: string } }[]
 }
 
-export const createDclNameChecker = (
+export async function createWorldNamePermissionChecker(
+  components: Pick<AppComponents, 'config' | 'ethereumProvider' | 'fetch' | 'logs' | 'marketplaceSubGraph'>
+): Promise<IWorldNamePermissionChecker> {
+  const logger = components.logs.getLogger('check-permissions')
+  const nameValidatorStrategy = await components.config.requireString('NAME_VALIDATOR')
+  switch (nameValidatorStrategy) {
+    case 'THE_GRAPH_DCL_NAME_CHECKER':
+      logger.info('Using TheGraph DclNameChecker')
+      return createTheGraphDclNameChecker(components)
+    case 'ON_CHAIN_DCL_NAME_CHECKER':
+      logger.info('Using OnChain DclNameChecker')
+      return await createOnChainDclNameChecker(components)
+    case 'ENDPOINT_NAME_CHECKER':
+      logger.info('Using Endpoint NameChecker')
+      return await createEndpointNameChecker(components)
+    case 'NOOP_NAME_CHECKER':
+      logger.info('Using NoOp NameChecker')
+      return await createNoOpNameChecker()
+
+    // Add more name validator strategies as needed here
+  }
+  throw Error(`Invalid nameValidatorStrategy selected: ${nameValidatorStrategy}`)
+}
+
+export const createTheGraphDclNameChecker = (
```

Comment:
nit: put these functions as async functions instead of const anonymous functions?